pkg/tfbridge/detailed_diff.go

```diff
@@ -350,79 +359,124 @@ func (differ detailedDiffer) makeListDiff(
 
 func computeSetHashChanges(
 	oldIdentities, newIdentities hashIndexMap,
-) (removed, added hashIndexMap) {
-	removed = hashIndexMap{}
-	added = hashIndexMap{}
+) (removed, added []arrayIndex) {
+	removed = []arrayIndex{}
+	added = []arrayIndex{}
 
 	for elementHash := range oldIdentities {
 		if _, ok := newIdentities[elementHash]; !ok {
-			removed[elementHash] = oldIdentities[elementHash]
+			removed = append(removed, oldIdentities[elementHash])
 		}
 	}
 
 	for elementHash := range newIdentities {
 		if _, ok := oldIdentities[elementHash]; !ok {
-			added[elementHash] = newIdentities[elementHash]
+			added = append(added, newIdentities[elementHash])
 		}
 	}
 
+	sort.Slice(removed, func(i, j int) bool {
+		return removed[i] < removed[j]
+	})
+	sort.Slice(added, func(i, j int) bool {
+		return added[i] < added[j]
+	})
+
 	return
 }
 
-func (differ detailedDiffer) matchNewIndicesToInputs(
-	path propertyPath, changedIdentities hashIndexMap,
-) hashIndexMap {
-	matched := hashIndexMap{}
-
+// matchPlanElementsToInputs is used to match the plan elements to the inputs.
+// It returns a map of inputs indices to the planned state indices.
+func (differ detailedDiffer) matchPlanElementsToInputs(
```

Comment:
This is a pre-existing condition but using differ.newInputs in this function instead of accepting it as a parameter does a dis-service to readability as that's a key logical parameter here.